extension/index.googleFunction.js

```diff
@@ -7,31 +7,57 @@ const logger = utils.getLogger()
 
 exports.extensionTrigger = async (request, response) => {
   try {
+    const obj = request?.body?.resource?.obj
+    if (!obj) {
+      return utils.sendGoogleFunctionResponse({
+        response,
+        statusCode: 400,
+        body: {
+          errors: [
+            {
+              code: 'BadRequest',
+              message: 'Invalid body payload.',
+            },
+          ],
+        },
+      })
+    }
     const authToken = auth.getAuthorizationRequestHeader(request)
-    const { obj } = request.body.resource
     const paymentResult = await paymentHandler.handlePayment(obj, authToken)
     if (paymentResult.success) {
-      response.status(200).send({
-        actions: paymentResult.data ? paymentResult.data.actions : [],
-      })
-    } else {
-      response.status(400).send({
-        errors: paymentResult.data ? paymentResult.data.errors : undefined,
+      return utils.sendGoogleFunctionResponse({
+        response,
+        statusCode: 200,
+        body: {
+          actions: paymentResult.data ? paymentResult.data.actions : [],
+        },
       })
     }
+    return utils.sendGoogleFunctionResponse({
+      response,
+      statusCode: 400,
+      body: {
+        errors: paymentResult.data ? paymentResult.data.errors : undefined,
+      },
+    })
   } catch (err) {
-    const errorMessage = `Unexpected error: ${JSON.stringify(
+    const errorMessage = `Unexpected error: ${err.message}`
+    const errorStackTrace = `Unexpected error: ${JSON.stringify(
       serializeError(err)
     )}`
-    logger.error(errorMessage)
+    logger.error(errorStackTrace)
 
-    response.status(500).send({
-      errors: [
-        {
-          code: 'InvalidOperation',
-          message: errorMessage,
-        },
-      ],
+    return utils.sendGoogleFunctionResponse({
+      response,
+      statusCode: 400,
+      body: {
+        errors: [
+          {
+            code: 'InvalidOperation',
+            message: errorMessage,
```

Comment:
Yes. we can change to `BadRequest` here
https://github.com/commercetools/commercetools-adyen-integration/pull/630/commits/1895bf5cfd68993a2ca29daa9330fa23cf6da7e7

Reply:
CTP doesn't accept `BadRequest` as error code, otherwise CTP throws exception to user.
In this exception, the stack-trace shows a list of error code it can support. `InvalidOperation` is one of them.